src/app/core/directives/server-html.directive.ts

```diff
@@ -99,7 +99,11 @@ export class ServerHtmlDirective implements AfterContentInit, AfterViewInit, OnD
         }
 
         if (cb && this.callbacks && typeof this.callbacks[cb] === 'function') {
+          // handle links with callback functions, e.g. <a callback="availableCallbackFunction">
           this.callbacks[cb]();
+        } else if (href.startsWith('#')) {
+          // handle fragment links / anchor navigation
+          this.router.navigate([this.router.url.split('#')[0]], { fragment: href.replace('#', '') });
```

Comment:
results in error-page.  
lets try without router:
`document.getElementById(href.replace('#', '')).scrollIntoView();`

Reply:
Changed to `.scrollIntoView` solution.